packages/react-router/src/components/Router/Router.tsx

```diff
@@ -20,11 +20,18 @@ export default function Router({location, children}: Props) {
     throw new Error(NO_LOCATION_ERROR);
   }
 
-  const locationString =
-    typeof location === 'object' ? location.href : location;
+  const locationObject =
+    typeof location === 'object'
+      ? {
+          pathname: location.pathname,
+          search: location.search,
+          hash: '',
+          state: undefined,
+        }
```

Comment:
Final thought:

The reason why this have to happen is this line 
[`location = { ...path }`](https://github.com/ReactTraining/history/blob/v4.10.1/modules/LocationUtils.js#L14)
where history is using spread operator to copy the object before using it, and we pass in URL object and copying it using spread operator means we lost all the property we actually do care about (like pathname & search).

Since the document recommend using [`{ pathname, search, hash, state }`](https://github.com/ReactTraining/react-router/blob/master/packages/react-router/docs/api/StaticRouter.md#location-object) instead of URL object. 

I am updating the type to reflect the only two pieces of information we need, and building an object to pass into StaticRouter that reflect this. 



This example below illustrate what it looks like using spread operator on URL object
```
> const url = new URL('https://gestalt-web-application-example.myshopify.io/test123');
undefined
> url
URL {
  href:
   'https://gestalt-web-application-example.myshopify.io/test123',
  origin: 'https://gestalt-web-application-example.myshopify.io',
  protocol: 'https:',
  username: '',
  password: '',
  host: 'gestalt-web-application-example.myshopify.io',
  hostname: 'gestalt-web-application-example.myshopify.io',
  port: '',
  pathname: '/test123',
  search: '',
  searchParams: URLSearchParams {},
  hash: '' }
> const foo = {...url};
undefined
> foo
{ [Symbol(context)]:
   URLContext {
     flags: 400,
     scheme: 'https:',
     username: '',
     password: '',
     host: 'gestalt-web-application-example.myshopify.io',
     port: null,
     path: [ 'test123' ],
     query: null,
     fragment: null },
  [Symbol(query)]: URLSearchParams {} }
> 
```